wgpu-hal/src/gles/web.rs

```diff
@@ -0,0 +1,296 @@
+        gl.shader_source(fragment, include_str!("./web/present.frag"));
+        gl.compile_shader(fragment);
+        gl.attach_shader(program, vertex);
+        gl.attach_shader(program, fragment);
+        gl.link_program(program);
+        gl.delete_shader(vertex);
+        gl.delete_shader(fragment);
+        gl.bind_texture(glow::TEXTURE_2D, None);
+
+        program
+    }
+}
+
+impl crate::Surface<super::Api> for Surface {
+    unsafe fn configure(
+        &mut self,
+        device: &super::Device,
+        config: &crate::SurfaceConfiguration,
+    ) -> Result<(), crate::SurfaceError> {
+        let gl = &device.shared.context.lock();
+
+        if let Some(swapchain) = self.swapchain.take() {
+            // delete all frame buffers already allocated
+            gl.delete_framebuffer(swapchain.framebuffer);
+        }
+
+        if self.present_program.is_none() {
+            self.present_program = Some(Self::create_present_program(gl));
+        }
+
+        if self.texture.is_none() {
+            self.texture = Some(gl.create_texture().unwrap());
+        }
+
+        let desc = device.shared.describe_texture_format(config.format);
+        gl.bind_texture(glow::TEXTURE_2D, self.texture);
+        gl.tex_parameter_i32(
+            glow::TEXTURE_2D,
+            glow::TEXTURE_MIN_FILTER,
+            glow::NEAREST as _,
+        );
+        gl.tex_parameter_i32(
+            glow::TEXTURE_2D,
+            glow::TEXTURE_MAG_FILTER,
+            glow::NEAREST as _,
+        );
+        gl.tex_storage_2d(
+            glow::TEXTURE_2D,
+            1,
+            desc.internal,
+            config.extent.width as i32,
+            config.extent.height as i32,
+        );
+
+        let framebuffer = gl.create_framebuffer().unwrap();
+        gl.bind_framebuffer(glow::READ_FRAMEBUFFER, Some(framebuffer));
+        gl.framebuffer_texture_2d(
+            glow::READ_FRAMEBUFFER,
+            glow::COLOR_ATTACHMENT0,
+            glow::TEXTURE_2D,
+            self.texture,
+            0,
+        );
+        gl.bind_texture(glow::TEXTURE_2D, None);
+
+        self.swapchain = Some(Swapchain {
+            extent: config.extent,
+            // channel: config.format.base_format().1,
+            format: config.format,
+            format_desc: desc,
+            framebuffer,
+        });
+        Ok(())
+    }
+
+    unsafe fn unconfigure(&mut self, device: &super::Device) {
+        let gl = device.shared.context.lock();
+        if let Some(swapchain) = self.swapchain.take() {
+            gl.delete_framebuffer(swapchain.framebuffer);
+        }
+        if let Some(renderbuffer) = self.texture.take() {
+            gl.delete_texture(renderbuffer);
+        }
+    }
+
+    unsafe fn acquire_texture(
+        &mut self,
+        _timeout_ms: u32,
+    ) -> Result<Option<crate::AcquiredSurfaceTexture<super::Api>>, crate::SurfaceError> {
+        let sc = self.swapchain.as_ref().unwrap();
+        let texture = super::Texture {
```

Comment:
Hi not sure if I used the PR incorrect but with the code until https://sotrh.github.io/learn-wgpu/beginner/tutorial3-pipeline/#what-s-a-pipeline I get this compiler error for this line of code:

```
error[E0063]: missing field `copy_size` in initializer of `gles::Texture`
   --> C:\Users\marce\.cargo\git\checkouts\wgpu-56857823b01a1582\9c69a96\wgpu-hal\src\gles\web.rs:279:23
    |
279 |         let texture = super::Texture {
    |                       ^^^^^^^^^^^^^^ missing `copy_size`

For more information about this error, try `rustc --explain E0063`.
error: could not compile `wgpu-hal` due to previous error
```

Reply:
After my most recent push this branch isn't currently working. I've still got to fix issues that appeared after rebasing on top of the latest master. I'll try to get to fixing this soon, but I've been pretty busy.